### PR TITLE
docs(aws service): Document credential expiration behavior

### LINKED ISSUE #7013: Vector should reread AWS_SHARED_CREDENTIALS_FILE on SIGHUP
### Vector Version

```
vector 0.12.1 (v0.12.1 x86_64-unknown-linux-gnu 2021-03-12)
```

### Vector Configuration File

```toml
[sinks.aws_bar_aws_s3]
  type = "aws_s3"
  bucket = "our_bucket"
  compression = "gzip"
  healthcheck = true
  region = "us-west-2"
  batch.max_size = 1049000000
  batch.timeout_secs = 60
  buffer.type = "memory"
  encoding.codec = "ndjson"
  filename_append_uuid = false
  filename_time_format =""
  key_prefix = "bar_foo/%Y/%m/%d/%H/%M/${HOSTNAME}"
  filename_extension = "0.json.gz"
  inputs = ["bar_aws_s3_source"]

Environment="AWS_SHARED_CREDENTIALS_FILE=/etc/vector/creds2"

# cat /etc/vector/creds2
[default]
AWS_SECRET_ACCESS_KEY=ZZZZZZZZZZZZZ
AWS_SESSION_TOKEN=REALLYLONGSESSIONKEYSTRING
AWS_ACCESS_KEY_ID=XXXXXXXXXXXXX
AWS_CREDENTIAL_EXPIRATION=2021-04-04T03:19:21.000Z

```

### PR DESCRIPTION
**Pull request by ktff**
Ref #7013

### CODE REVIEW

File: docs/reference/components/aws.cue [hunk 1]
```diff
@@ -186,6 +186,12 @@ components: _aws: {
 						account access.
 						"""
 				},
+				{
+					title: "Expiration"
+					body: """
+						Vector will again search for AWS credentials once they expire.
```
Comment (jszwedko) [inline docs/reference/components/aws.cue:192]: I might just put this in the top "AWS Authentication" section. If they expire, it checks all credential mechanisms, yes?

Comment (ktff) [reply docs/reference/components/aws.cue:192]: Yes, it rechecks as if it's fetching them for the first time.

File: docs/reference/components/aws.cue [hunk 2]
```diff
@@ -166,6 +166,8 @@ components: _aws: {
 
 				If credentials are not found the [healtcheck](#healthchecks) will fail and an
 				error will be [logged][docs.monitoring#logs].
+
+				Vector will again search for AWS credentials once they expire.
```
Comment (jszwedko) [inline docs/reference/components/aws.cue:170]:
```suggestion
				Vector will search again for AWS credentials once they expire.
```

**Review by jszwedko — APPROVED** [review]
Thanks!

**Pull request merged**